contracts/TokenConverter/AbstractTokenConverter.sol

```diff
@@ -0,0 +1,635 @@
+
+        /// amount of tokenAddressOut after including incentive
+        uint256 conversionWithIncentive = MANTISSA_ONE + configuration.incentive;
+        /// conversion rate after considering incentive(conversionWithIncentive)
+        uint256 tokenInToOutConversion = (tokenInUnderlyingPrice * conversionWithIncentive) / tokenOutUnderlyingPrice;
+
+        amountOutMantissa = (amountInMantissa * tokenInToOutConversion) / EXP_SCALE;
+        amountConvertedMantissa = amountInMantissa;
+
+        /// If contract has less liquity for tokenAddressOut than amountOutMantissa
+        if (maxTokenOutReserve < amountOutMantissa) {
+            amountConvertedMantissa = ((maxTokenOutReserve * EXP_SCALE) / tokenInToOutConversion);
+            amountOutMantissa = maxTokenOutReserve;
+        }
+    }
+
+    /// @notice To get the amount of tokenAddressIn tokens sender would send on receiving amountOutMantissa tokens of tokenAddressOut
+    /// @param amountOutMantissa Amount of tokenAddressOut user wants to receive
+    /// @param tokenAddressIn Address of the token to convert
+    /// @param tokenAddressOut Address of the token to get after convert
+    /// @return amountConvertedMantissa Amount of tokenAddressOut should be transferred after convert
+    /// @return amountInMantissa Amount of the tokenAddressIn sender would send to contract before convert
+    /// @custom:error InsufficientInputAmount error is thrown when given input amount is zero
+    /// @custom:error ConversionConfigNotEnabled is thrown when convert is disabled or config does not exist for given pair
+    function getAmountIn(
+        uint256 amountOutMantissa,
+        address tokenAddressIn,
+        address tokenAddressOut
+    ) public returns (uint256 amountConvertedMantissa, uint256 amountInMantissa) {
+        if (amountOutMantissa == 0) {
+            revert InsufficientInputAmount();
+        }
+
+        ConversionConfig memory configuration = convertConfigurations[tokenAddressIn][tokenAddressOut];
+
+        if (!configuration.enabled) {
+            revert ConversionConfigNotEnabled();
+        }
+
+        priceOracle.updateAssetPrice(tokenAddressIn);
+        priceOracle.updateAssetPrice(tokenAddressOut);
+
+        uint256 maxTokenOutReserve = balanceOf(tokenAddressOut);
+        uint256 tokenInUnderlyingPrice = priceOracle.getPrice(tokenAddressIn);
+        uint256 tokenOutUnderlyingPrice = priceOracle.getPrice(tokenAddressOut);
+
+        /// amount of tokenAddressOut after including incentive
+        uint256 conversionWithIncentive = MANTISSA_ONE + configuration.incentive;
+        /// conversion rate after considering incentive(conversionWithIncentive)
+        uint256 tokenInToOutConversion = (tokenInUnderlyingPrice * conversionWithIncentive) / tokenOutUnderlyingPrice;
+
+        amountInMantissa = ((amountOutMantissa * EXP_SCALE) / tokenInToOutConversion);
+        amountConvertedMantissa = amountOutMantissa;
```

Comment:
I think this can be moved to the `else` block of the following `if` statement, to avoid unnecessary calculations

Reply:
I have added this fix in the PVE003 fix: https://github.com/VenusProtocol/protocol-reserve/pull/17/commits/92342eb81d4436cc0fa9a88d92b07615bebee562